crates/glaredb/src/lib.rs

```diff
@@ -18,38 +29,81 @@ use derive_builder::Builder;
+/// instance, and are an analogue to the commandline arguments to
+/// produce a "running database". The ConnectOptionsBuilder provides a
+/// chainable interface to construct these values and produce as
+/// structure. You can construct `ConnectOptions` fully without the
+/// builder interface, depending on your preference.
+///
+/// The `.connect()` method on `ConnectionOptions` is the main way to
+/// produce a GlareDB instance. All instances use an in-process
+/// metastore (catalog).
+///
+/// The `connection_target`, `location`, `spill_path` and
+/// `storage_options` all control the behavior of a local, single node
+/// instance, while the remaining options configure a GlareDB instance
+/// for hybrid-execution.
 #[derive(Default, Builder)]
 pub struct ConnectOptions {
+    /// The connection_target specifies where the GlareDB instances
+    /// storage is. This is either, in memory (unspecified or
+    /// `memory://`), a path to something on the local file-system for
+    /// persistent local databases, or an object-store URL for
+    /// databases that are backed up onto cloud storage.
     #[builder(setter(into))]
     pub connection_target: Option<String>,
+    /// Location is that path **within** the `connection_target` where
+    /// the database's files will be stored. This is required for all
+    /// object-store backed GlareDB instances and ignored in all other
+    /// cases.
     #[builder(setter(into))]
     pub location: Option<String>,
+    /// Specifies the location on the local file system where this
+    /// process will write files so that it can spill data for
+    /// operations to local disk (sorts, large joins, etc.)
     #[builder(setter(into))]
     pub spill_path: Option<String>,
+    /// Defines the options used to configure the object store
+    /// (credentials, etc.)
     #[builder(setter(into), default = "HashMap::new()")]
     pub storage_options: HashMap<String, String>,
 
+    /// By default, the client will connect to the GlareDB service
+    /// using TLS. When this option is specified (and true), the this
+    /// GlareDB instance will establish an insecure connection. Use for
+    /// testing and development.
     #[builder(setter(strip_option))]
     pub disable_tls: Option<bool>,
+    /// Location of the GlareDB clout instance used by GlareDB
+    /// negotiate out-of-band certificate provisioning.
     #[builder(default = "Some(\"https://console.glaredb.com\".to_string())")]
     #[builder(setter(into, strip_option))]
     pub cloud_addr: Option<String>,
+    /// Client type distinguishes what kind of remote client this is,
+    /// and is used for logging and introspection.
     #[builder(default = "Some(RemoteClientType::Cli)")]
     #[builder(setter(strip_option))]
     pub client_type: Option<RemoteClientType>,
+    /// Specify an optional environment reader, which GlareDB uses in
+    /// embedded cases so that queries can bindings to extract tables
```

Comment:
This sentence is a bit off, not 100% sure what was intended, maybe something like:

```
can bind to extracted tables
from variables in scope of data frames or the
output of a query
```